### PR TITLE
refined functions to get packages indices

### PR DESCRIPTION
**Pull request by rgildein**
Add the _get_package_indices, which is searching all "**/Packages*" in dists, but instead of picking up only the first path, it's filtering a path from the same parent directory.

### CODE REVIEW

**Comment by rgildein** [comment]
blocked by: #27

**Review by jneo8 — APPROVED** [review]
LGTM

**Review by chanchiwai-ray — APPROVED** [review]
LGTM for the improvement to get package indices; and also thanks for enhance the code quality a bit.

**Review by agileshaw — APPROVED** [review]
LGTM

File: tests/unit/test_utils.py [hunk 1]
```diff
@@ -49,6 +49,60 @@ def test_clean_packages(packages, exp_result):
...
+)
+def test_get_package_indices(tmp_path, paths, exp_indices):
+    """Test helper function to obtain package indices from dists directory."""
+    dists = tmp_path / "dists"
+    for path in paths:
+        _path = dists / path
+        _path.mkdir(parents=True, exist_ok=True)
+        _path.touch()
+
+    assert utils._get_package_indices(dists) == [dists / indice for indice in exp_indices]
```
Comment (dashmage) [inline tests/unit/test_utils.py]: nit: Should be `[dists / index for index in exp_indices]` since index is the singular for indices.

**Review by dashmage — APPROVED** [review]
Left a small comment. Really nice PR and LGTM!

File: tests/unit/test_utils.py (hunk 1 above)
Comment (rgildein) [reply tests/unit/test_utils.py]: I'm afraid that you are right, pushing my changes.

**Review by jneo8 — APPROVED** [review]
LGTM

**Pull request merged**